### PR TITLE
Update README to include all configs for cluster agent

### PR DESCRIPTION
**Pull request by kelseyma**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,13 +89,28 @@ spec:
...
 | `account`                 |  AppDynamics Account Name                                    |       Required             |
-| `image` | Cluster Agent image reference | During the Beta period access to the image must be requested from AppDynamics|
-| `resources` |  Definitions of resources and limits for the machine agent  | See resource recommendations below |
-| `appName` |  Name of the cluster  | Required |
+| `image` | Cluster Agent image reference | Required |
+| `resources` |  Definitions of resources and limits for the Cluster Agent | See resource recommendations below |
+| `appName` |  Name of the Cluster Agent | Required |
+| `nsToMonitor` | List of namespaces the Cluster Agent should initially monitor | Default is `default` |
+| `eventUploadInterval` | Interval in seconds at which events will be uploaded | Default 10 sec |
+| `containerRegistrationInterval` | Interval in seconds at which a container should be registered | Default 120 sec |
```
Comment (sashaPM) [inline README.md]: Which container are you referring to?

File: README.md [hunk 2]
```diff
@@ -89,13 +89,28 @@ spec:
...
+| `httpClientTimeout` | HTTP client timeout value in seconds | Default 30 sec |
```
Comment (sashaPM) [inline README.md]: Not cleat what this HTTP client is doing

File: README.md [hunk 3]
```diff
@@ -89,13 +89,28 @@ spec:
 | ------------------------- | ------------------------------------------------------------ | -------------------------- |
 | `controllerUrl`           |  Url of the AppDynamics controller                            |       Required             |
 | `account`                 |  AppDynamics Account Name                                    |       Required             |
-| `image` | Cluster Agent image reference | During the Beta period access to the image must be requested from AppDynamics|
-| `resources` |  Definitions of resources and limits for the machine agent  | See resource recommendations below |
-| `appName` |  Name of the cluster  | Required |
+| `image` | Cluster Agent image reference | Required |
```
Comment (sashaPM) [inline README.md]: Are we planning to have an official image?

File: README.md [hunk 4]
```diff
@@ -89,13 +89,28 @@ spec:
...
+| `resources` |  Definitions of resources and limits for the Cluster Agent | See resource recommendations below |
+| `appName` |  Name of the Cluster Agent | Required |
+| `nsToMonitor` | List of namespaces the Cluster Agent should initially monitor | Default is `default` |
+| `eventUploadInterval` | Interval in seconds at which events will be uploaded | Default 10 sec |
```
Comment (sashaPM) [inline README.md]: Are you referring to Kubernetes events?

File: README.md [hunk 5]
```diff
@@ -89,13 +89,28 @@ spec:
...
+| `containerRegistrationInterval` | Interval in seconds at which a container should be registered | Default 120 sec |
+| `httpClientTimeout` | HTTP client timeout value in seconds | Default 30 sec |
+| `proxyUrl` | Url of the proxy server (`protocol://domain:port`) | Not set by default | 
+| `proxyUser` | Name of the user that is authenticated by the proxy host | Not set by default |
 | `metricsSyncInterval` | Sampling interval at which metrics should be collected periodically | Default 30 sec |
```
Comment (sashaPM) [inline README.md]: I assume that metrics are reported are collected in real time. Do we want to say that it is how frequently they are reported to the controller?

File: README.md [hunk 6]
```diff
@@ -89,13 +89,28 @@ spec:
...
+| `clusterMetricsSyncInterval` | Sampling interval at which cluster level metrics should be collected periodically | Default 60 sec |
+| `metadataSyncInterval` | Sampling interval at which metadata should be collected periodically | Default 60 sec |
```
Comment (sashaPM) [inline README.md]: Which metadata? Can you elaborate?

File: README.md [hunk 7]
```diff
@@ -89,13 +89,28 @@ spec:
...
+| `containerFilter` | Definitions of whitelisted/blacklisted names and blacklisted labels to filter | See container filter explanation below |
+| `containerBatchSize` | Batch size used in container registration | Default 25 sec |
```
Comment (sashaPM) [inline README.md]: Not clear what container registration you are referring to. Also it is about size, but the default is in sec.

File: README.md [hunk 8]
```diff
@@ -89,13 +89,28 @@ spec:
...
+| `containerParallelRequestLimit` | Max parallel requests to controller for container registration | Default is 3 |
+| `podBatchSize` | Batch size used in pod registration | Default is 30 |
```
Comment (sashaPM) [inline README.md]: Can you elaborate what it means. Also what are the units ? Records?

File: README.md [hunk 9]
```diff
@@ -89,13 +89,28 @@ spec:
...
+| `metricUploadRetryCount` | Number of times metric upload action to be attempted if unsuccessful the first time | Default is 3 |
+| `metricUploadRetryIntervalMilliSeconds` | Interval between consecutive metric upload retries, in milli seconds | Default is 5 |
 | `logLevel` | Logging level (`INFO`, `DEBUG`, `WARN`, `TRACE`) | Default `INFO` |
-| `logFileSizeMb` | The maximum file size of the log in MB. | Default is 5 |
-| `logFileBackups` | The maximum number of backups the log saves. When the maximum number of backups is reached, the 3 oldest log file after the initial log file is deleted. | Default is 3 |
+| `logFileSizeMb` | Maximum file size of the log in MB | Default is 5 |
+| `logFileBackups` | Maximum number of backups the log saves. When the maximum number of backups is reached, the oldest log file after the initial log file is deleted. | Default is 3 |
+| `stdoutLogging` | Write logging info to stdout | Default is `"true"` |
```
Comment (sashaPM) [inline README.md]: So, if this attribute is omitted, the agent will log to console? Can you pls double-check.

File: README.md [hunk 10]
```diff
@@ -109,7 +124,24 @@ Example resource limits:
       memory: "100M"
 ```
 
-
+Example container filter:
```
Comment (sashaPM) [inline README.md]: I recommend giving an example of the entire spec. A minimalistic spec and then a more fleshed out with all attributes that you want to highlight (like filters). Snippets like below are not readily usable

File: README.md (hunk 3 above)
Comment (kelseyma) [reply README.md]: No, customers have to build their own image from the distribution package provided via the downloads portal

File: README.md (hunk 4 above)
Comment (kelseyma) [reply README.md]: These uploaded events are actually ones that we generate from the Kubernetes events.

File: README.md (hunk 1 above)
Comment (kelseyma) [reply README.md]: The containers that are being monitored by the Cluster Agent

File: README.md (hunk 10 above)
Comment (kelseyma) [reply README.md]: Added spec in new commit

File: README.md (hunk 7 above)
Comment (kelseyma) [reply README.md]: fixed default and updated description

File: README.md (hunk 9 above)
Comment (kelseyma) [reply README.md]: Updated description to be more clear

File: README.md (hunk 8 above)
Comment (kelseyma) [reply README.md]: Updated description and added units

File: README.md (hunk 6 above)
Comment (kelseyma) [reply README.md]: Added details about metadata

File: README.md (hunk 5 above)
Comment (kelseyma) [reply README.md]: Updated to be more clear

File: README.md (hunk 2 above)
Comment (kelseyma) [reply README.md]: Updated description

File: README.md [hunk 11]
```diff
@@ -77,40 +77,42 @@ metadata:
...
-| `proxyUrl` | Url of the proxy server (`protocol://domain:port`) | Not set by default | 
-| `proxyUser` | Name of the user that is authenticated by the proxy host | Not set by default |
-| `metricsSyncInterval` | Sampling interval at which metrics should be collected periodically | Default 30 sec |
-| `clusterMetricsSyncInterval` | Sampling interval at which cluster level metrics should be collected periodically | Default 60 sec |
-| `metadataSyncInterval` | Sampling interval at which metadata should be collected periodically | Default 60 sec |
-| `containerFilter` | Definitions of whitelisted/blacklisted names and blacklisted labels to filter | See container filter explanation below |
-| `containerBatchSize` | Batch size used in container registration | Default 25 sec |
-| `containerParallelRequestLimit` | Max parallel requests to controller for container registration | Default is 3 |
-| `podBatchSize` | Batch size used in pod registration | Default is 30 |
+| `eventUploadInterval` | Interval in seconds at which events are uploaded to the Controller | Default 10 sec |
```
[the remark this answers is not in the document]
Comment (kelseyma) [reply README.md]: Updated description to match the official GA docs

**Pull request merged**